src/main/java/io/rpg/view/popups/DialoguePopup.java

```diff
@@ -0,0 +1,81 @@
+package io.rpg.view.popups;
+
+import io.rpg.viewmodel.DialoguePopupViewModel;
+import javafx.event.EventHandler;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Group;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.paint.Color;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class DialoguePopup extends Scene {
+
+  private final DialoguePopupViewModel viewModel;
+  private final List<String> textPages;
+  private int currentPage = 0;
+
+  public DialoguePopup(String text, Image npcImage, String backgroundPath, String buttonPath) {
+    this(text, npcImage);
+    viewModel.setBackgroundImage(backgroundPath);
+    viewModel.setCloseButtonImage(buttonPath);
+  }
+
+  public DialoguePopup(String text, Image npcImage) {
+    super(new Group(), Color.TRANSPARENT);
+    FXMLLoader loader = new FXMLLoader(Objects.requireNonNull(DialoguePopupViewModel.class.getResource("dialogue-popup-view.fxml")));
+    Parent root = null;
+
+    try {
+      root = loader.load();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    this.setRoot(root);
+
+    viewModel = loader.getController();
+    viewModel.setNpcImage(npcImage);
+    viewModel.setBackgroundImage("file:assets/popup-background.png"); //images do not load from fxml on my computer
+    viewModel.setCloseButtonImage("file:assets/close-button.png");
+    viewModel.setNextButtonImage("file:assets/right-arrow.png");
+    viewModel.setPreviousButtonImage("file:assets/left-arrow.png");
+    viewModel.setNpcFrameImage("file:assets/npc-frame.png");
+    this.setFill(Color.TRANSPARENT);
+
+    textPages = List.of(text.split("(?<=\\G.{200})")); //split text into 200-letter pages
```

Comment:
splitting question into pages is nice, but we'll need a limit for answer's length, cause dividing answers into pages seems like too much for me (though it's just my opinion)

Reply:
For now I've decided not to implement a back and forth dialogue. Thought of it as more of a guide or a quick tip for a player. I guess that's still up for debate.